aea/helpers/dialogue/base.py

```diff
@@ -581,7 +571,7 @@ def update(self, message: Message) -> Optional[Dialogue]:
             dialogue = self._create_opponent_initiated(
                 dialogue_opponent_addr=message.counterparty,
                 dialogue_reference=dialogue_reference,
-                role=Dialogue.role_from_first_message(message),
```

Comment:
That was wrongly implemented, the concrete/derived class is needed..Fixing this had a lot of downstream effects